tmux/tmux
# Add options and arguments to display-menu to set menu border lines

This PR adds border line styling to `display-menu` and is a follow-up PR of #2947.

ℹ️ This PR builds upon the proposed changes from #3649; to only see the relevant changes please see https://github.com/afh/tmux/compare/afh-menu-border-style...afh:tmux:afh-menu-border-lines

To test:
* Build and run tmux `./tmux -vv -f/dev/null -Ltest`
* Set `menu-border-lines` options and display menu (see `test_menu_border_lines.sh` below)
* Use `-b` argument when displaying a menu (see `test_menu_border_lines.sh ` below)

👉 **NOTA BENE:** There are several `TODO:` comments in the code that should to be clarified prior applying the changes.

ℹ️ The `none` style is currently the same as `padded` due to difficulties in removing the border completely, for details see [this comment on #2947](https://github.com/tmux/tmux/pull/2947#issuecomment-1195668262)

ℹ️ The changes in [`tty-acs.c`](https://github.com/tmux/tmux/pull/3650#diff-0061931f0e579d624769a3e92854fbd0db819ee849c4c75d74928eb0114c2abb) fix an issue with the rounded characters which used the heavy ([U+2523](https://unicodeplus.com/U+2523) and [U+252B](https://unicodeplus.com/U+252B)) instead of the light ([U+251C](https://unicodeplus.com/U+251C) and [U+2524](https://unicodeplus.com/U+2524)) variant of the "Box Drawings * Vertical And (Right|Left)" character.  

<details><summary><code> test_menu_border_lines.sh </code></summary>

```sh
#!/bin/ksh

clear
echo "# Expect default tmux border lines menu when unstyled"
echo "% ./tmux set -gwU menu-border-lines"
./tmux set -gwU menu-border-lines
echo "% ./tmux menu -T Default lines '' '' -bar '' '' '' baz '' ''"
./tmux menu -T Default lines '' '' -bar '' '' '' baz '' ''

clear
lines=heavy
echo "# Expect tmux $lines menu border lines when set via tmux options, e.g.:"
echo "% ./tmux set -gw menu-border-lines $lines"
./tmux set -gw menu-border-lines $lines
echo "% ./tmux menu -T $lines lines '' '' -bar '' '' '' baz '' ''"
./tmux menu -T $lines lines '' '' -bar '' '' '' baz '' ''

for lines in double single rounded heavy ; do
clear
echo "# Expect tmux $lines menu border lines when set via tmux arguments overriding previously set options, e.g.:"
echo "% ./tmux menu -T $lines -b $lines lines '' '' -bar '' '' '' baz '' ''"
./tmux menu -T $lines -b $lines lines '' '' -bar '' '' '' baz '' ''
done

clear
lines=padded
echo "# Expect tmux $lines menu border lines when set via tmux arguments overriding previously set options, e.g.:"
echo "% ./tmux menu -T $lines -b $lines -S bg=green lines '' '' -bar '' '' '' baz '' ''"
./tmux menu -T $lines -b $lines -S bg=green lines '' '' -bar '' '' '' baz '' ''

clear
lines=none
echo "# Expect tmux $lines menu border lines when set via tmux arguments overriding previously set options, e.g.:"
echo "% ./tmux menu -T $lines -b $lines -S bg=orange lines '' '' -bar '' '' '' baz '' ''"
./tmux menu -T $lines -b $lines -S bg=orange lines '' '' -bar '' '' '' baz '' ''
```

</details>

| | |
| --- | --- |
| Default | <img width=90% src=https://github.com/tmux/tmux/assets/16507/e323f2bb-b565-4cc4-89f6-bf22a09dd041>
| Heavy | <img width=90% src=https://github.com/tmux/tmux/assets/16507/0a5783d2-7756-496d-a409-12b4926c5ed3)>
| Double | <img width=90% src=https://github.com/tmux/tmux/assets/16507/7841af59-1da7-433d-a1cb-7be450b524e2> <img width=90% src=https://github.com/tmux/tmux/assets/16507/e15059b2-ab8f-4643-85d9-46572a4e6667>
| Rounded | <img width=90% src=https://github.com/tmux/tmux/assets/16507/852ac5ac-f6e1-4109-9196-e8cf46546231>
| Padded | <img width=90% src=https://github.com/tmux/tmux/assets/16507/e6f9c676-6646-4acd-a666-b81c8e58aff8>
| None | <img width=90% src=https://github.com/tmux/tmux/assets/16507/6845385f-b5bc-4306-8403-b5e670fc8fb1>
